### ISSUE TITLE
Custom privacy policies / terms for each instance

### ISSUE THREAD

**Issue by StaticallyTypedRice** [LemmyNet/lemmy#721]
To my knowledge, many jurisdictions (like the EU) require websites to have a comprehensive privacy policy, and in general having one is just a good idea. My suggestion would be to have a page that administrators can edit in order to describe their server’s privacy practices.

We could also provide a default one for all instances, but chances are that to stay accurate, each instance will have to edit it. This should also be made clear to instance administrators.

**Comment by dioraman** [comment]
I second this.
Mastodon adopts [default terms of service + privacy policy](### https://mastodon.social/terms) for any instance. It is originally adapted from [Discourse](https://www.discourse.org/privacy) but I think the Mastodon basis can be improved.

**Comment by dessalines** [comment]
Does mastodon / pleroma have an editable privacy policy?

**Comment by dioraman** [comment]
I knew Mastodon and Pleroma instances with modified privacy policies.

**Comment by nathgit** [comment]
In general, I'm very hesitant to sign up to any site without reading their privacy policy first, (& I would hope others would feel the same) so I hope one is coming. At least since this one doesn't use an email address, there's no chance of them being able to sell your email address to companies that will spam you. The downside to that is if you forget your password, there's no way to recover it. Email aliases and/or filters are good ways to fight spam too though. Reddit doesn't require an email address either from what I've heard but that doesn't mean there aren't still potential privacy issues. There's a site that lists all publicly available knowledge about any reddit username you type in, all derived from info they've posted, comments they made, communities they follow, etc;. The amount of info you can find is unbelievable.

**Comment by dessalines** [comment]
> There's atleast 2 sites that list all publicly available knowledge about any reddit username you type in, all derived from info they've posted, comments they made, communities they follow, etc;. The amount of info you can find is unbelievable.

There isn't a way to prevent that unfortunately for publicly available data, I'm sure there's tools to do this for fediverse users too. But when we add private communities, then that'll be a way to restrict your viewable content.

**Comment by nathgit** [comment]
I know. I just meant that people put a lot of private data on social networks (which they shouldn't), so a privacy policy is needed.

**Comment by nathgit** [comment]
Is there any sort of functioning privacy policy template? I'm considering signing up but generally don't sign up on sites without a privacy policy. Not that I don't trust it, just being careful especially since the account can't be deleted.

**Comment by nathgit** [comment]
Here's a couple links that might be helpful:
https://security.berkeley.edu/how-write-effective-website-privacy-statement
https://www.privacypolicies.com/blog/how-write-privacy-policy/

**Comment by Nutomic** [comment]
We are hosted in the EU so a privacy policy for the US wont help us.

**Comment by dessalines** [comment]
Ya. Anyone who wants to make this, I'd suggest starting with mastodon's as a template that @dioraman linked.

**Comment by nathgit** [comment]
> To my knowledge, many jurisdictions (like the EU) require websites to have a comprehensive privacy policy, and in general having one is just a good idea. 
California has [calOPPA](https://consumercal.org/about-cfc/cfc-education-foundation/california-online-privacy-protection-act-caloppa-3/), which actually took effect in '04 but has been amended since requires any site operating in California that collects user info to have a privacy policy explaining how they use that info & mandates certain things that must be included in the privacy policy, like an outline of exactly what info is collected.

**Comment by nathgit** [comment]
> We are hosted in the EU so a privacy policy for the US wont help us.

Here's a couple sites Specific to GDPR privacy policies. The first explains how to make one, the second is a FREE generator.
[GDPR privacy policy](https://gdpr.eu/privacy-notice/)
[Generator](https://www.freeprivacypolicy.com/)
The generator creates a privacy policy, a cookie policy, & several others compatible with GDPR, calOPPA, Google Analytics, etc;

**Comment by nicfab** [comment]
I second this.
I was thinking of creating an ad hoc web page with my privacy policy reachable from my Lemmy instance.
It would be much more appreciable to have a default Lemmy space/box or page to fill with policy contents.

**Comment by nathgit** [comment]
There should be some sort of privacy policy. I was admittedly a little confused by the seeming lack of interest when I listed the links I did above allowing you to easily generate a privacy policy. I take my privacy seriously & I read every site's privacy policy before signing up so I know what is being done with my data, so a site not having one is a red flag for me. At the very least I would use an email alias, which is a good idea anyway, in case of a site being hacked, or sold.

**Comment by nicfab** [comment]
@Nutomic Thank you very much.
I appreciate your commit.
From my perspective, the title "Legal information" is not wrong, but you can also think of shortening it to only "Legal."
I agree with you on the other considerations related to the content; each admin will fill the space with the appropriate content.
I would also highlight that the content should at least be terms and conditions and privacy policy.
For example, as I read, in the [Mastodon "legalese" default document](https://mastodon.nicfab.it/terms) (from my Mastodon instance), there is a section related to data retention for IP addresses that they set, by default, in 90 days.
An admin can know that information only from developers.
In conclusion, besides the appreciable commit, the admin should also know some technical information from the developers. The admin will provide adequate information to the users to guarantee that nobody else apart from the sysadmin can access that information only for technical purposes.
The sysadmin, obviously, has to adopt any security measure on his server.

**Comment by Nutomic** [comment]
About IP addresses, Lemmy doesnt store them at all. They are only stored in log files, but that also happens when you serve static html files from your server.

**Comment by BanzooIO** [comment]
I feel this is pretty important to the survival of Lemmy. I AM NOT A LAWYER, but I have created a template based on the Mastodon privacy policy if anyone wants a basic framework to start from:

https://github.com/BanzooIO/federated_policies_and_tos/blob/main/lemmy-privacy-policy.md

I am not overly experienced with instance management yet, but I have done my best to cover all aspects of how data is shared. Please contribute in correcting any errors.

I also feel it is important for admins to disclose the lack of SSL support in connecting to PostgreSQL and what the local admin has done to mitigate the risk.

**Comment by nicfab** [comment]
If it can be helpful for you and the project, here is the privacy policy on my Lemmy instance: https://community.nicfab.it/legal
Furthermore, here is the privacy policy on my Mastodon instance: https://mastodon.nicfab.it/privacy-policy
Both are works in progress, and they might be modified anytime.

**Comment by BanzooIO** [comment]
> If it can be helpful for you and the project, here is the privacy policy on my Lemmy instance: https://community.nicfab.it/legal Furthermore, here is the privacy policy on my Mastodon instance: https://mastodon.nicfab.it/privacy-policy Both are works in progress, and they might be modified anytime.

Awesome, nice to see an admin actually on this. Thanks. Definitely open to all input on this, and as you've correctly deduced from the title, plan to expand into other platforms. How would you prefer to be attributed (or if you would prefer not).

As my current version is going to maybe be a bit unsettling for uninformed users, I have created an additional optional policy introduction: https://github.com/BanzooIO/federated_policies_and_tos/blob/main/optional-privacy-policy-intro.md
